sim/core/attack.go

```diff
@@ -409,6 +418,16 @@ func (unit *Unit) EnableAutoAttacks(agent Agent, options AutoAttackOptions) {
 			baseDamage := spell.Unit.RangedWeaponDamage(sim, spell.RangedAttackPower(target))
 			spell.CalcAndDealDamage(sim, target, baseDamage, spell.OutcomeRangedHitAndCrit)
 		},
+
+		ExtraCastCondition: func(sim *Simulation, target *Unit) bool {
+			// ranged has minimum range of 5
+			return unit.DistanceFromTarget > 5
+		},
+	}
+
+	// set character at ranged if primary attack is ranged
+	if options.AutoSwingRanged && unit.DistanceFromTarget == 0 {
+		unit.DistanceFromTarget = 5.1
 	}
```

Comment:
I dont think this should be set here. You can overwrite the current distance from target from just disabling/enabling auto attacks now. This default should be probably set per spec during initialization of the spec

Reply:
Also thats why we have to Move to Distance action supporting prepull. So users can set their prefered starting distance during prepull

Reply:
Oh missed that. Though this method would only be called once on sim init. Will have a look to get that done more properly

Reply:
Another thing is it's gonna be pretty ugly to add
```
		ExtraCastCondition: func(sim *Simulation, target *Unit) bool {
			// ranged has minimum range of 5
			return unit.DistanceFromTarget > 5
		},
```
To every Hunter spell. I'd want both MinRange and MaxRange to be part of the SpellConfig if we're gonna have movement and boss api imo, much cleaner

Reply:
> Oh missed that. Though this method would only be called once on sim init. Will have a look to get that done more properly

This method actually can be called during sim as well (some spells disable/enable autoattacks)